src/go/rdctl/cmd/shell.go

```diff
@@ -82,22 +82,22 @@ func doShellCommand(cmd *cobra.Command, args []string) error {
 		}
 
 		for _, distroName := range distroNames {
-			if !checkWSLIsRunning(distroName) {
-				continue
+			err = assertWSLIsRunning(distroName)
+			if err == nil {
+				commandName = "wsl"
+				args = append([]string{
+					"--distribution", distroName,
+					"--exec", "/usr/local/bin/wsl-exec",
+				}, args...)
+				found = true
+				break
 			}
-
-			commandName = "wsl"
-			args = append([]string{
-				"--distribution", distroName,
-				"--exec", "/usr/local/bin/wsl-exec",
-			}, args...)
-			found = true
-			break
 		}
 
 		if !found {
 			// We did not find a running distribution that we can use.
 			// No further output wanted, so just exit with the desired status.
+			fmt.Fprintf(os.Stderr, "%s", err)
```

Comment:
I guess it is a matter of personal preference, but why not

```suggestion
			fmt.Fprint(os.Stderr, err.String())
```